sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -451,4 +451,14 @@ class SQLConfSuite extends QueryTest with SharedSparkSession {
     val e2 = intercept[ParseException](sql("set time zone interval 19 hours"))
     assert(e2.getMessage contains "The interval value must be in the range of [-18, +18] hours")
   }
+
+  test("SPARK-34454: configs from the legacy namespace should be internal") {
+    val nonInternalLegacyConfigs = spark.sessionState.conf.getAllDefinedConfs
+      .filter { case (key, _, _, _) => key.contains(".legacy.") }
```

Comment:
To be more clear, `.legacy.` -> `spark.sql.legacy.`?

Reply:
AFAIK spark core also has legacy confs, but not sure `spark.sessionState.conf.getAllDefinedConfs` returns spark core confs.

Reply:
`getAllDefinedConfs` returns only sql configs

Reply:
@dongjoon-hyun If you can guarantee that the legacy namespace can be only inside of `spark.sql.` and not in any nested namespaces, let's change it.

> To be more clear that this is about SQL configs ...

@dongjoon-hyun It is pretty clear now. The test checks that any legacy configs are internal. But now, you say no-no - not any, only one `spark.sql.legacy` because somehow you guarantee that nobody adds a nested legacy namespace like `spark.sql.super_feature.legacy.*`.

Reply:
I believe `spark.sql.legacy` is the reserved namespace for all legacy and we had better block those `.legacy.` addition in this PR (if you are worry about that).
> But now, you say no-no - not any, only one spark.sql.legacy because somehow you guarantee that nobody adds a nested legacy namespace like spark.sql.super_feature.legacy.*.

Reply:
Or, do you really need to put `.legacy.` in a random position in the config namespace?

Reply:
I just want to catch the cases too. Even for now, we have only `spark.sql.legacy` in OSS. Let's imagine there is a Spark vendor which has its own configs like `spark.databricks.legacy.*` ;-) It would be nice if the test works with their configs too. Also, maybe in the future, someone will add a nested `legacy` namespace, so, the test will be ready for that.

Reply:
Are there any non-`spark.sql.legacy` legacy conf in Apache Spark codebase?

Reply:
Okay, but I agree that `spark.sql.legacy` is explicit and covers all cases in Apache Spark codebase. I will change like that.